RecoMuon/L2MuonSeedGenerator/src/L2MuonSeedGeneratorFromL1T.cc

```diff
@@ -0,0 +1,383 @@
+            LogDebug(metname) << debug.dumpMuonId(newTSOSDet->geographicalId());
+
+            if (newTSOS.isValid()){
+
+              //LogDebug(metname) << "(x, y, z) = (" << newTSOS.globalPosition().x() << ", " 
+              //                  << newTSOS.globalPosition().y() << ", " << newTSOS.globalPosition().z() << ")";
+              LogDebug(metname) << "pos: (r=" << newTSOS.globalPosition().mag() << ", phi=" 
+                                << newTSOS.globalPosition().phi() << ", eta=" << newTSOS.globalPosition().eta() << ")";
+              LogDebug(metname) << "mom: (q*pt=" << newTSOS.charge()*newTSOS.globalMomentum().perp() << ", phi=" 
+                                << newTSOS.globalMomentum().phi() << ", eta=" << newTSOS.globalMomentum().eta() << ")";
+
+              //LogDebug(metname) << "State on it";
+              //LogDebug(metname) << debug.dumpTSOS(newTSOS);
+
+               //PTrajectoryStateOnDet seedTSOS;
+              edm::OwnVector<TrackingRecHit> container;
+  
+              if(useOfflineSeed) {
+                const TrajectorySeed *assoOffseed = 
+                  associateOfflineSeedToL1(offlineSeedHandle, offlineSeedMap, newTSOS);
+
+                if(assoOffseed!=0) {
+                  PTrajectoryStateOnDet const & seedTSOS = assoOffseed->startingState();
+                  TrajectorySeed::const_iterator 
+                    tsci  = assoOffseed->recHits().first,
+                    tscie = assoOffseed->recHits().second;
+                  for(; tsci!=tscie; ++tsci) {
+                    container.push_back(*tsci);
+                  }
+                  output->push_back(L2MuonTrajectorySeed(seedTSOS,container,alongMomentum,
+                                    MuonRef(muColl,  distance(muColl->begin(muColl->getFirstBX()),it)  )));
+
+                }
+                else {
+                  if(useUnassociatedL1) {
+                    // convert the TSOS into a PTSOD
+                    PTrajectoryStateOnDet const & seedTSOS = trajectoryStateTransform::persistentState( newTSOS,newTSOSDet->geographicalId().rawId());
+                    output->push_back(L2MuonTrajectorySeed(seedTSOS,container,alongMomentum,
+                                      MuonRef(muColl,  distance(muColl->begin(muColl->getFirstBX()),it)  )));
+                  }
+                } 
+              }  
+              else {
+                // convert the TSOS into a PTSOD
+                PTrajectoryStateOnDet const & seedTSOS = trajectoryStateTransform::persistentState( newTSOS,newTSOSDet->geographicalId().rawId());
+                output->push_back(L2MuonTrajectorySeed(seedTSOS,container,alongMomentum,
+                                  MuonRef(muColl,  distance(muColl->begin(muColl->getFirstBX()),it)  )));
+              }
+        
+            }
+          }
+        }
+      }
+  
+    }
+  } else {
+    edm::LogWarning("MissingProduct") << "L1Upgrade muon bx collection not found." << std::endl;
```

Comment:
The product should always be present in the event if the path/sequence is appropriately configured.
I find that it's more problematic to let the code to keep going if the only expected product is missing.
hltGmtStage2Digis should always produce the product (even empty).